extensions/ql-vscode/src/vscode-tests/no-workspace/query-results.test.ts

```diff
@@ -237,19 +248,85 @@ describe('query-results', () => {
-          interpretedResultsPath: '/d/e/f'
+  describe('splat and slurp', () => {
+    // TODO also add a test for round trip starting from file
+    it('should splat and slurp query history', async () => {
+      const infoSuccessRaw = createMockFullQueryInfo('a', createMockQueryWithResults(false, false, '/a/b/c/a', false));
+      const infoSuccessInterpreted = createMockFullQueryInfo('b', createMockQueryWithResults(true, true, '/a/b/c/b', false));
+      const infoEarlyFailure = createMockFullQueryInfo('c', undefined, true);
+      const infoLateFailure = createMockFullQueryInfo('d', createMockQueryWithResults(false, false, '/a/b/c/d', false));
+      const infoInprogress = createMockFullQueryInfo('e');
+      const allHistory = [
+        infoSuccessRaw,
+        infoSuccessInterpreted,
+        infoEarlyFailure,
+        infoLateFailure,
+        infoInprogress
+      ];
+
+      const mockDbManager = {
+        findDatabaseItem: (uri: Uri) => {
+          return allHistory.find(info => info.initialInfo.databaseInfo.databaseUri === uri.fsPath)?.completedQuery?.query.dbItem;
+        }
+      } as DatabaseManager;
+      const allHistoryPath = path.join(queriesDir, 'all-history.json');
+      await FullQueryInfo.splat(allHistory, allHistoryPath);
+      const allHistoryActual = await FullQueryInfo.slurp(allHistoryPath, mockDbManager, mockConfig);
+
+      // the dispose methods will be different. Ignore them.
+      allHistoryActual.forEach(info => {
+        if (info.completedQuery) {
+          const completedQuery = info.completedQuery;
+          (completedQuery as any).dispose = undefined;
+
+          // these fields should be missing on the slurped value
+          // but they are undefined on the original value
+          if (!('logFileLocation' in completedQuery)) {
+            (completedQuery as any).logFileLocation = undefined;
+          }
+          const query = completedQuery.query;
+          if (!('quickEvalPosition' in query)) {
+            (query as any).quickEvalPosition = undefined;
+          }
+          if (!('templates' in query)) {
+            (query as any).templates = undefined;
+          }
         }
-      } as QueryEvaluationInfo,
+      });
+      allHistory.forEach(info => {
+        if (info.completedQuery) {
+          (info.completedQuery as any).dispose = undefined;
+        }
+      });
+
+      // make the diffs somewhat sane by comparing each element directly
+      // also, the db items are non enumerable. Must test them explicitly
+      for (let i = 0; i < allHistoryActual.length; i++) {
+        expect(allHistoryActual[0]).to.deep.eq(allHistory[0]);
+        const info = allHistoryActual[i];
+        if (info.completedQuery) {
+          const dbItem = info.completedQuery.query.dbItem;
+          expect(dbItem).to.deep.eq(allHistory[i].completedQuery!.query.dbItem);
+        }
+      }
+      expect(allHistoryActual.length).to.deep.eq(allHistory.length);
+    });
+  });
```

Comment:
This test will ensure that whatever we serialize can be properly deserialized, but as you can see, it's complex.